static/js/components/UploadResumeView.js

```diff
@@ -0,0 +1,168 @@
+import { ALLOWED_FILE_EXTENTIONS } from "../constants"
+import { getFilenameFromPath } from "../util/util"
+import type { ApplicationDetail } from "../flow/applicationTypes"
+
+
+const getUploadedFileMassage = (resume_filepath: string) => {
+  return `Your uploaded resume: ${getFilenameFromPath(resume_filepath)}`
+}
+
+const done_btn_style = {
+  color: "#fff",
+  backgroundColor: "#d9534f",
+  borderColor: "#d43f3a",
+}
+
+type UploaderProps = {
+  application: ApplicationDetail
+}
+export const ResumeUploader = (props: UploaderProps) => {
+  const { application } = props
+  const getUploadParams = () => {
+    return {
+      url: `/api/applications/${application.id}/resume/`,
+      ...DEFAULT_POST_OPTIONS
+    }
+  }
+
+  const handleChangeStatus = ({ meta, file }, status) => {
+    const upload_message_el = document.getElementsByClassName('uploaded-resume')[0]
+    if (!upload_message_el)
+      return
+    let upload_status
+    switch (status){
+      case 'error_upload':
+        upload_status = "Upload Failed"
+        break
+      case 'done':
+        upload_status = "Uploaded successfully!"
+        break
+      case 'removed':
+        upload_status = getUploadedFileMassage(application.resume_filepath)
+        break
+      default:
+        upload_status = ""
+    }
+    upload_message_el.innerHTML = upload_status
+  }
+  const handleSubmit = (files, allFiles) => {
+    allFiles.forEach(f => f.remove())
+  }
+
+  const inputContent = () => {
+    return <div className="input-label">
+      <h3>Drag file here or click to Browse</h3>
+      {application.resume_filepath? <p> This will replace your existing resume.</p> : null}
+      <p>Allowed file types: {join(",", keys(ALLOWED_FILE_EXTENTIONS))}</p>
+    </div>
+  }
+
+  return (
+    <Dropzone
+      getUploadParams={getUploadParams}
+      onChangeStatus={handleChangeStatus}
+      inputContent={inputContent}
+      onSubmit={handleSubmit}
+      submitButtonContent="Done"
+      classNames="drop-outline"
+      maxFiles={1}
+      styles={{ dropzone: { minHeight: 200, maxHeight: 250 }, submitButton: done_btn_style }}
+      accept={join(",", values(ALLOWED_FILE_EXTENTIONS))}
+    />
+  )
+}
+type Props = {
+  application: ?ApplicationDetail,
+  uploadResume: (linkedinUrl: string, applicationId: number) => Promise<void>
+}
+
+export class UploadResumeView extends React.Component<Props> {
+  render() {
+    const { uploadResume, application } = this.props
+
+    if (!application) {
+      return null
+    }
+    return (
+      <div className="container">
+        <div className="drawer-wrapper">
+          <h2 className="drawer-title">Resume or LinkedIn Profile</h2>
+          <p>Please upload resume or a link to your LinkedIn profile</p>
+          <h3>Resume</h3>
+          <ResumeUploader application={application} />
+
+          {application.resume_filepath ? (
+            <p className="uploaded-resume">
+              {getUploadedFileMassage(application.resume_filepath)}
```

Comment:
Sorry about the earlier file name suggestion. I forgot that the file names are altered when we save them. We'll probably want to save the original filenames upon upload, but that doesn't need to be part of this PR. For the time being, I can't imagine we're going to want the altered filename being showed here. Can you reimplement the split function you had here before, and just make sure the filename still shows correctly if it has underscores in it?